command/ca/provisioner/add.go

```diff
@@ -107,38 +106,50 @@ Use the flag multiple times to configure multiple service accounts.`,
 				Name: "gcp-project",
 				Usage: `The Google project <id> used to validate the identity tokens.
 Use the flag multipl etimes to configure multiple projects`,
-			},
-			cli.StringFlag{
-				Name:  "azure-tenant",
-				Usage: `The Microsoft Azure tenant <id> used to validate the identity tokens.`,
-			},
-			cli.StringSliceFlag{
-				Name: "azure-resource-group",
-				Usage: `The Microsoft Azure resource group <name> used to validate the identity tokens.
-Use the flag multipl etimes to configure multiple resource groups`,
 			},
 			cli.DurationFlag{
 				Name: "instance-age",
 				Usage: `The maximum <duration> to grant a certificate in AWS and GCP provisioners.
 A <duration> is sequence of decimal numbers, each with optional fraction and a
 unit suffix, such as "300ms", "-1.5h" or "2h45m". Valid time units are "ns",
 "us" (or "µs"), "ms", "s", "m", "h".`,
-			},
-			cli.BoolFlag{
-				Name: "disable-custom-sans",
-				Usage: `On cloud provisioners, if anabled only the internal DNS and IP will be added as a SAN.
-By default it will accept any SAN in the CSR.`,
-			},
-			cli.BoolFlag{
-				Name: "disable-trust-on-first-use,disable-tofu",
-				Usage: `On cloud provisioners, if enabled multiple sign request for this provisioner
-with the same instance will be accepted. By default only the first request
-will be accepted.`,
 			},
 			cli.BoolFlag{
 				Name:  "ssh",
 				Usage: `Enable SSH on the new provisioners.`,
 			},
+			cli.StringFlag{
+				Name:  "type",
+				Value: provisioner.TypeJWK.String(),
+				Usage: `The <type> of provisioner to create. Type is a case-insensitive string
+and must be one of:
+    **JWK**
+    : Uses an JWK key pair to sign provisioning tokens. (default)
+
+    **OIDC**
+    : Uses an OpenID Connect provider to sign provisioning tokens.
+
+    **AWS**
+    : Uses Amazon AWS instance identity documents.
+
+    **GCP**
+    : Use Google instance identity tokens.
+
+    **Azure**
+    : Uses Microsoft Azure identity tokens.
+
+    **ACME**
+    : Uses the ACME protocol to create certificates.
+
+    **X5C**
+    : Uses an X509 Certificate / private key pair to sign provisioning tokens.
+`,
+			},
```

Comment:
type should be one of the first flag to appear if not the first. The order in this list is important.

Reply:
Make sure related flags are next to each other.

Reply:
Why is it important that type be one of the first flags? I sorted the flags alphabetically because I think it was pretty chaotic before that. We have hundreds of lines of flag definitions and it was hard to know if the flag you cared about was already defined. 

Is there any reason to keep related flags next to one another (other than that they are functionally related)? We should discuss coming up with a consistent pattern, because from my perspective it was chaos.